qiskit/algorithms/linear_solvers/observables/linear_system_observable.py

```diff
@@ -0,0 +1,74 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An abstract class for linear systems solvers in Qiskit's aqua module."""
+from abc import ABC, abstractmethod
+from typing import Union, Optional, List
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.opflow import PauliSumOp
+
+
+class LinearSystemObservable(ABC):
+    """An abstract class for linear system observables in Qiskit."""
+    @abstractmethod
+    def observable(self, num_qubits: int) -> Union[PauliSumOp, List[PauliSumOp]]:
+        """The observable operator.
+
+        Args:
+            num_qubits: The number of qubits on which the observable will be applied.
+
+        Returns:
+            The observable as a sum of Pauli strings.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def post_rotation(self, num_qubits: int) -> Union[QuantumCircuit, List[QuantumCircuit]]:
+        """The observable circuit.
+
+        Args:
+            num_qubits: The number of qubits on which the observable will be applied.
+
+        Returns:
+            The observable as a QuantumCircuit.
```

Comment:
Is this description it mentions nothing of post rotation. Can the description be improved in any way - its not so very different from the `observable` above. Given the description the method name to me could easily be `observable_circuit` since rather than getting an operator back I am getting ciircuit.

Reply:
changed to `observable_circuit`